File: frontend/src/client/service/occupancy-map.dto.ts
export type OccupancyMapPublic = {
  id: string
  name: string
  pgm_file_path: string
  yaml_file_path: string
  created_at: string
  updated_at: string
}

export type OccupancyMapsPublic = {
  data: Array<OccupancyMapPublic>
  count: number
}

File: backend/src/main/java/com/gistacsl/mosaic/occupancymap/dto/OccupancyMapDto.java
package com.gistacsl.mosaic.occupancymap.dto;

import java.time.OffsetDateTime;
import java.util.UUID;

public class OccupancyMapDto {

    public record Res(
            UUID id,
            String name,
            String pgmFilePath,
            String yamlFilePath,
            OffsetDateTime createdAt,
            OffsetDateTime updatedAt
    ) {
    }
}